iso15118/secc/controller/simulator.py

```diff
@@ -1069,24 +1107,33 @@ async def get_15118_ev_certificate(
         except Exception as exc:
             raise Exception(f"Error creating signature {exc}")
 
-        header = MessageHeaderV2(
-            session_id=cert_install_req.header.session_id,
-            signature=signature,
-        )
-        body = Body.parse_obj({"CertificateInstallationRes": cert_install_res.dict()})
-        to_be_exi_encoded = V2GMessageV2(header=header, body=body)
-        exi_encoded_cert_installation_res = EXI().to_exi(
-            to_be_exi_encoded, Namespace.ISO_V2_MSG_DEF
-        )
+        if isinstance(cert_install_req, SupportedAppProtocolReq) or isinstance(
+            cert_install_req, SupportedAppProtocolRes
+        ):
+            logger.info(f"Ignoring EXI decoding of a {type(cert_install_req)} message.")
+            return ""
+        else:
```

Comment:
This if block is not required. Everything in the else condition is required though. :)

Reply:
@shalinnijel2 : `cert_install_req` is derived from the` EXI().from_exi()` function, which has the potential to yield various data types, some of which may lack the associated header field. This absence could result in a program error when accessed. Consequently, removing the conditional filter for variable type could lead to mypy raising an error, specifically stating the absence of a "header" field.

Reply:
@santiagosalamandri - ok, understood. Would something like this be slightly better?
So we still do type checking to satisfy mypy as follows:
```
        if isinstance(cert_install_req, CertificateInstallationReq):
            header = MessageHeaderV2(
                session_id=cert_install_req.header.session_id,
                signature=signature,
            )
            body = Body.parse_obj(
                {"CertificateInstallationRes": cert_install_res.dict()}
            )
            to_be_exi_encoded = V2GMessageV2(header=header, body=body)
            exi_encoded_cert_installation_res = EXI().to_exi(
                to_be_exi_encoded, Namespace.ISO_V2_MSG_DEF
            )

            # base64.b64encode in Python is a binary transform
            # so the return value is byte[]
            # But the CPO expects exi_encoded_cert_installation_res
            # as a string, hence the added .decode("utf-8")
            base64_encode_cert_install_res = base64.b64encode(
                exi_encoded_cert_installation_res
            ).decode("utf-8")

            return base64_encode_cert_install_res
        else:
            logger.info(f"Ignoring EXI decoding of a {type(cert_install_req)} message.")
            return ""
```